### DISCUSSION TITLE
cross-repo commits/pull requests

### DISCUSSION THREAD

**Discussion by ngbtech** [community/community#34967] [New to GitHub]
Is what is the best straightforward way to create a script that would basically commit the same branch name from the head of one repo to another repo?

**Comment by airtower-luna** [comment]
What exactly are you trying to achieve? If you just want to have the same branch with the exact same commits in two remote repositories (on GitHub or otherwise) you can simply add both as remotes and push to both of them. See: [Git Basics - Working with Remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes)

**Reply by ngbtech** [reply]
I am trying to keep my dev, acceptance and production environments in separate repos, and have an automated push or pull from one environment's repo to another.  You're right, I could do this on the servers themselves using bash scripts with git commands and will probably end up there, but I was hoping to keep this more centralized if possible.   If that's not an option, I'll implement what you suggested.

**Reply by airtower-luna** [reply]
Well, now I'm wondering why you'd ever want to keep those in separate repositories. Wouldn't branches in one repository do? :sweat_smile:

**Reply by ngbtech** [reply]
That's what I have always done in the past, but with the team I have I would like to keep them separate this time if possible.

**Comment by spenserblack** [comment]
Assuming that the histories between repositories are related (https://github.com/community/community/discussions/34967#discussioncomment-3789383 implies they are) then you can add a remote and merge from that.
For example, if you're trying to merge commits from a repository called "foo" into a repository called "bar":

```bash
# in bar

# initial setup
git remote add foo https://host.com/user/foo.git

# assuming your primary branch is main
git fetch foo
git merge foo/main
```
This is a pretty common way to keep a fork in sync with its upstream repo, for example.

It sounds like these repositories are effectively forks of each other, so you can possibly just do that, but having multiple forks with the same owner only works for organizations AFAIK.

But, personally, I agree with @airtower-luna about using branches. I'd use [branch protections](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule) instead.
